### PR TITLE
Close HF2 connection before reloading window

### PR DESCRIPTION
**Pull request by caitlin-hennessy**

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -1029,11 +1029,24 @@ export class ProjectView
...
-            if (!r) return;
-            this.reload = true; //Indicate we are goint to reload next.
-            workspace.resetAsync()
-                .done(() => window.location.reload(),
-                () => window.location.reload())
+            if (!r) return Promise.resolve();
+            if (hf2Connection) {
+                return hf2Connection.disconnectAsync()
+                .then(() => {
+                    this.reload = true; //Indicate we are going to reload next.
```
Comment (riknoll) [inline webapp/src/app.tsx]: Move this code block into a function so that it doesn't need to be duplicated below

Comment (abchatra) [reply webapp/src/app.tsx]: +1

**Comment by caitlin-hennessy** [comment]
@riknoll Look better now?

**Comment by riknoll** [comment]
What's with the commit from @microsoftsam?

**Comment by pelikhan** [comment]
Isn't this fix required for Adafruit?

**Comment by caitlin-hennessy** [comment]
@peli This PR should be deleted, it was redone here https://github.com/Microsoft/pxt/pull/3353

**Pull request closed without merging**